ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -4151,16 +4152,18 @@ public boolean dropPartition(String dbName, String tableName, List<String> parti
       }
 
       if (nParts > nBatches * batchSize) {
-       String validWriteIdList = null;
-       Long tableId = null;
-       if (AcidUtils.isTransactionalTable(tbl)) {
-        ValidWriteIdList vWriteIdList = getValidWriteIdList(tbl.getDbName(), tbl.getTableName());
-        validWriteIdList = vWriteIdList != null ? vWriteIdList.toString() : null;
-        tableId = tbl.getTTable().getId();
-      }
+        String validWriteIdList = null;
```

Comment:
nit: I wonder if we can use the following:
```java
 GetPartitionsByNamesRequest req = MetaStoreUtils.convertToGetPartitionsByNamesRequest(tbl.getDbName(), tbl.getTableName(),
            partNames.subList(nBatches*batchSize, nParts), getColStats, Constants.HIVE_ENGINE, null,
            null);
 List<org.apache.hadoop.hive.metastore.api.Partition> tParts = getPartitionsByNames(req, tbl);

Reply:
We need to push down the `vaildWriteIdList` and the `tableId`, so if there is caching in the catalog side then we the cache could be refreshed when needed